### PR TITLE
improve movement in filemanager layout

### PR DESCRIPTION
**Pull request by Solarer**
This PR improves the behavior of the cursor when using arrow keys without mouseover in filemanager layout and also provides a bugfix.

**Use arrow keys without selection/mouseover**

Current behavior:
- Starting with Up / Left arrow key is ignored (nothing happens)
- Starting with Right arrow key skips first image and starts on image 2
- Starting with Down arrow key skips first row and starts in row 2

New behavior:
- any arrow key starts at image 1 in row 1. Then we start moving.

**Page up/down button when at start/end of collection**

Old behavior:
- do nothing (highlighted image stays the same)

New behavior:
- page up jumps to first image of collection if we already are in row 1
- page down jumps to last image of collection if we would scroll past the last image

Bugfix: To reproduce zoom to 4 columns in filemanager layout and move the cursor to any of the images 2-7. When clicking page-up button the cursor disappeared. This has been fixed.

### CODE REVIEW

**Comment by Solarer** [comment]
Not sure why one of the checks fails. Can someone help me debug?

**Comment by johnny-bit** [comment]
simple :)

```
/home/runner/work/darktable/darktable/src/src/dtgtk/thumbtable.c:2529:3: error: unknown type name ‘bool’
 2529 |   bool first_move = (baseid <= 0);
      |   ^~~~
```

don't use `bool`, but gboolean

File: src/dtgtk/thumbtable.c [hunk 1]
```diff
@@ -2526,57 +2526,79 @@ static gboolean _filemanager_key_move(dt_thumbtable_t *table, dt_thumbtable_move
 {
   // base point
   int baseid = dt_control_get_mouse_over_id();
+  bool first_move = (baseid <= 0);
```
Comment (johnny-bit) [inline src/dtgtk/thumbtable.c:2529]: don't use `bool` in C code since it's not universally supported type

**Comment by Solarer** [comment]
Thanks, I was wondering why people were using that all the time.

**Comment by TurboGit** [comment]
Same comment as the culling PR, please squash both commits. Here as you can see we have an intermediate commit that won't compile properly and will hinder a bisecting session. TIA.

File: src/dtgtk/thumbtable.c [hunk 2]
```diff
@@ -2526,57 +2526,79 @@ static gboolean _filemanager_key_move(dt_thumbtable_t *table, dt_thumbtable_move
...
+    }
   }
-  // direct start/end
-  else if(move == DT_THUMBTABLE_MOVE_START)
-    newrowid = 1;
-  else if(move == DT_THUMBTABLE_MOVE_END)
-    newrowid = maxrowid;
 
-  if(newrowid == baserowid) return FALSE;
+  //if(newrowid == baserowid) return FALSE;
```
Comment (AlicVB) [inline src/dtgtk/thumbtable.c:2601]: is that the "fix" part of your PR (I don't reproduce your "cursor" hiding with master)
Anyway, if that's the case, better remove the line for real. git will keep the memory of it if needed !

**Review by AlicVB — COMMENTED** [review]
Seems good for me. Thanks for this
just merge your 2 commit, as @TurboGit explain... rebasing inside PR is not a problem

File: src/dtgtk/thumbtable.c (hunk 2 above)
Comment (Solarer) [reply src/dtgtk/thumbtable.c:2601]: No, looks like I broke something in git 😅. Will clean it up

**Comment by Solarer** [comment]
@AlicVB done

**Review by TurboGit — CHANGES_REQUESTED** [review]
This needs a conflict resolution, be careful this one is probably not trivial. TIA.

**Comment by Solarer** [comment]
All conflicts should be resolved now!

**Review by TurboGit — APPROVED** [review]
Thanks! Nice usability fixes.

**Comment by elstoc** [comment]
@Solarer please can you make sure that your commits are more descriptive. This commit appeared in master as "rebase with master and fix conflicts" which says nothing about what it does.

**Comment by TurboGit** [comment]
@elstoc : You're right, and I'm a bit faulty there too as I haven't looked at the commit log :(

**Pull request merged**